orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -95,7 +95,9 @@ class ORTTrainer(object):
             Inputs to the combined PyTorch model are concatenation of the :py:attr:`model`'s input and :py:attr:`loss_fn`'s label input.
             Outputs of the combined PyTorch model are concatenation of :py:attr:`loss_fn`'s loss output and :py:attr:`model`'s outputs.
         options (ORTTrainerOptions, default is None): options for additional features.
-
+        session_option: the session option instance for training session. if it is None, the default instance will be used.
+        provider_names: the sequence of providers used in the training session.
+        provider_options: the sequence of dict, in the order of providers registered to training session.
```

Comment:
(I commented this yesterday, but it seems to have been detached from the code...)

Can provider_options be a dict of dicts, and remove the need for a separate provider_names whose order needs to be kept in sync? (e.g. `{"CPUExecutionProvider": {"option1": "foo", "option2": "bar", ...}, ...}`)

Reply:
i think that will be cleaner. the reason i list them separately is just to match our python training session api:
https://github.com/microsoft/onnxruntime/blob/9835b46a1df5c1dfeb634c61668c644390577d47/orttraining/orttraining/python/deprecated/training_session.py#L15

Reply:
`orttrainer_options.py` is recommended for adding advanced and optional configuration for `ORTTrainer`.

The constructor only has the main required components for training.

Dic of dicts can easily be supported inside `ORTTrainerOptions` class